craft_providers/pro.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright 2021-2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""Pro-related module."""
+
+import json
+import logging
+
+import requests
+
+from craft_providers.errors import MachineTokenError
+
+logger = logging.getLogger(__name__)
+
+
+class ProToken:
+    """Class for retrieving and processing pro tokens."""
```

Comment:
Is there a specific reason to have a class here? Feels like it should just be a few free functions

Reply:
Fixed